File: flut/flutter/painting/basic_types.py
from flut._flut.engine import FlutEnumObject


class Axis(FlutEnumObject):
    horizontal: "Axis"
    vertical: "Axis"


class AxisDirection(FlutEnumObject):
    up: "AxisDirection"
    right: "AxisDirection"
    down: "AxisDirection"
    left: "AxisDirection"


class VerticalDirection(FlutEnumObject):
    up: "VerticalDirection"
    down: "VerticalDirection"

File: flut/.flutter/lib/flutter/painting/basic_types.dart
import 'package:flutter/painting.dart';
import 'package:flut/flut/object.dart';

class FlutAxis extends FlutEnumObject<Axis> {
  const FlutAxis()
    : super('Axis', const {
        'horizontal': Axis.horizontal,
        'vertical': Axis.vertical,
      });
}

class FlutAxisDirection extends FlutEnumObject<AxisDirection> {
  const FlutAxisDirection()
    : super('AxisDirection', const {
        'up': AxisDirection.up,
        'right': AxisDirection.right,
        'down': AxisDirection.down,
        'left': AxisDirection.left,
      });
}

class FlutVerticalDirection extends FlutEnumObject<VerticalDirection> {
  const FlutVerticalDirection()
    : super('VerticalDirection', const {
        'up': VerticalDirection.up,
        'down': VerticalDirection.down,
      });
}
